src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -32,15 +46,13 @@ const listStyles = ({ theme }) => css`
   `};
 `;
 
-const ButtonGroupList = styled('ul')`
-  ${listStyles};
-`;
+const ButtonGroupList = styled('ul')(baseStyles, alignmentStyles);
 
 /**
  * Groups its Button children into a list and adds margins between.
  */
-const ButtonGroup = ({ children }) => (
-  <ButtonGroupList>
+const ButtonGroup = ({ children, align }) => (
+  <ButtonGroupList align={align}>
```

Comment:
Maybe you could already introduce a spread for all other prop here? `({ children, ...props })` and then `<ButtonGroupList {...props} />`. This way stuff like `className` etc. will get propagated. Something the original implementation missed already. Not sure how to handle this in the `propTypes` though.